cpp/include/raft/linalg/map.cuh

```diff
@@ -22,119 +22,74 @@
 
 #include <raft/core/device_mdspan.hpp>
 #include <raft/core/device_resources.hpp>
-#include <raft/util/input_validation.hpp>
-#include <thrust/tabulate.h>
 
-namespace raft {
-namespace linalg {
-
-/**
- * @brief CUDA version of map
- * @tparam InType data-type upon which the math operation will be performed
- * @tparam MapOp the device-lambda performing the actual operation
- * @tparam TPB threads-per-block in the final kernel launched
- * @tparam Args additional parameters
- * @tparam OutType data-type in which the result will be stored
- * @param out the output of the map operation (assumed to be a device pointer)
- * @param len number of elements in the input array
- * @param map the device-lambda
- * @param stream cuda-stream where to launch this kernel
- * @param in the input array
- * @param args additional input arrays
- */
-template <typename InType,
-          typename MapOp,
-          typename IdxType = std::uint32_t,
-          int TPB          = 256,
-          typename OutType = InType,
-          typename... Args>
-void map_k(
-  OutType* out, IdxType len, MapOp map, cudaStream_t stream, const InType* in, Args... args)
-{
-  detail::mapImpl<InType, OutType, IdxType, MapOp, TPB, Args...>(
-    out, len, map, stream, in, args...);
-}
+namespace raft::linalg {
 
 /**
  * @defgroup map Mapping ops
  * @{
  */
 
 /**
- * @brief CUDA version of map
- * @tparam InType data-type for math operation of type raft::device_mdspan
- * @tparam MapOp the device-lambda performing the actual operation
- * @tparam TPB threads-per-block in the final kernel launched
- * @tparam OutType data-type of result of type raft::device_mdspan
- * @tparam Args additional parameters
- * @param[in] handle raft::device_resources
- * @param[in] in the input of type raft::device_mdspan
- * @param[out] out the output of the map operation of type raft::device_mdspan
- * @param[in] map the device-lambda
- * @param[in] args additional input arrays
+ * @brief Map a function over zero or more input mdspans.
+ *
+ * Usage example:
+ * @code{.cpp}
+ *  auto input = raft::make_device_vector<int>(res, n);
```

Comment:
These examples should be copy/paste for users without them having to dig through a bunch of headers to get them to compile. We should add the headers in here too. 

Can we also mention that this supports vectorized loads?